File: superfriend-py/src/superfriend/constant/chat_mode.py
class ChatMode:
    LITE_TASK = "lite-task"
    MEDIUM_TASK = "medium-task"
    COMPLEX_TASK = "complex-task"
    MCP = "mcp"
    AI = "ai"
    DEFAULT = LITE_TASK

    @classmethod
    def is_valid(cls, mode: str | None) -> bool:
        if mode is None:
            return False
        return (
            cls.LITE_TASK == mode
            or cls.MEDIUM_TASK == mode
            or cls.COMPLEX_TASK == mode
            or cls.MCP == mode
            or cls.AI == mode
        )

    @classmethod
    def normalize(cls, mode: str | None) -> str:
        if not cls.is_valid(mode):
            return cls.DEFAULT
        return mode or cls.DEFAULT

File: src/main/java/com/superfriend/superfriend/constant/ChatMode.java
package com.superfriend.superfriend.constant;

public final class ChatMode {

    private ChatMode() {
    }

    public static final String LITE_TASK = "lite-task";
    public static final String MEDIUM_TASK = "medium-task";
    public static final String COMPLEX_TASK = "complex-task";
    public static final String MCP = "mcp";
    public static final String AI = "ai";
    public static final String DEFAULT = LITE_TASK;

    public static boolean isValid(String mode) {
        if (mode == null) {
            return false;
        }
        return LITE_TASK.equals(mode)
            || MEDIUM_TASK.equals(mode)
            || COMPLEX_TASK.equals(mode)
            || MCP.equals(mode)
            || AI.equals(mode);
    }

    public static String normalize(String mode) {
        if (mode == null || !isValid(mode)) {
            return DEFAULT;
        }
        return mode;
    }
}
